src/PostgREST/App.hs

```diff
@@ -280,10 +280,10 @@ augumentRequestWithJoin schema allRels request =
 
 buildReadRequest :: [Relation] -> ApiRequest -> Either Text ReadRequest
 buildReadRequest allRels apiRequest  =
-  augumentRequestWithJoin schema rels =<< first formatParserError (foldr addFilter <$> (addOrder <$> readRequest <*> ord) <*> flts)
+  augumentRequestWithJoin schema rels
+  =<< first formatParserError (foldr addFilter <$> (foldr addOrder <$> readRequest <*> ords) <*> flts)
```

Comment:
as a sidenote, this is what's going in this line
`ReadRequest` is a tree that comes out of parsing the &select parameter.
Then we first add to that tree the `order` parameters, then we add `filters` and at the end, we look at the "relations" between the tree nodes and we add the necessary filters/conditions in order for the joins to work.
At the end we have a tree that we can just turn into a query, all the data is in it